Assets/Scripts/LevelManager.cs

```diff
@@ -0,0 +1,50 @@
+﻿using UnityEngine;
+using UnityEngine.SceneManagement;
+
+// TODO: determine best practice for level managers (all static members? always use FindObjectOfType<LevelManager>()? always use inspector variable?)
+public class LevelManager : MonoBehaviour
+{
+    
+    public UIManager ui;
+    
+    private bool levelOver;
+    private Rigidbody playerBody;
+    
+    private void Start()
+    {
+        levelOver = false;
+        playerBody = GameObject.FindGameObjectWithTag("Player").GetComponent<Rigidbody>();
+    }
+
+    private void Update()
+    {
+        if (levelOver && Input.anyKeyDown || Input.GetKeyDown(KeyCode.R))
+        {
+            SceneManager.LoadScene(SceneManager.GetActiveScene().buildIndex);
+        }
+    }
+
+    public void Win()
+    {
+        if (levelOver) return;
+        ui.SetLevelStatus("You win!");
+        levelOver = true;
+        playerBody.constraints = RigidbodyConstraints.FreezeAll;
+    }
+
+    public void Lose()
+    {
+        if (levelOver) return;
+        ui.SetLevelStatus("Game over!");
```

Comment:
I'd make SetLevelStatus take in a bool (maybe eventually an enum?) for the state the level is in. Then in UI manager (within the SetLevelStatus function), you can include all the presentation logic.